wp-content/plugins/tapestry/controller/class.tapestry-controller.php

```diff
@@ -3,4 +3,83 @@
  * Add/update/retrieve Tapestry post and its child nodes
  * 
  */
-class TapestryController {}
\ No newline at end of file
+class TapestryController {
+    /**
+     * Retrieve a Tapestry post
+     * 
+     * @param  Number @postId The Tapestry postId
+     * @return Object Tapestry
+     */
+    public function getTapestry($postId = null) {
+        // TODO: Use $this->postId that's passed in the constructor
+        // TODO: uncomment the two lines below for error handling
+        // after the PR for saving tapestry is merged.
+        /*
+        if (!$this->_isValidPostId($postId)) {
+            return $this->throwsError('INVALID_POST_ID');
+        }
+        */
+        $tapestry = $this->_getTapestryById($postId);
+        return $tapestry;
+    }
+
+    private function _getTapestryById($postId) {
+        $tapestry = get_post_meta($postId, 'tapestry', true);
+
+        $metadatas = array_map(function($nodeId) {
+            return get_metadata_by_mid('post', $nodeId);
+        }, $tapestry->nodes);
+
+        $nodeDatas = array_map(function($metadata) {
```

Comment:
I would append the content of this function into the previous `array_map`, rather than doing `array_map` again here since you are not using `$metadatas` anywhere else. i.e. just have 1 `array_map` rather than 2.

Reply:
Ah right. Yeah, it's a good point!